src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9784,7 +9800,6 @@ private bool IsEndOfDeclarationClause()
             switch (this.CurrentToken.Kind)
             {
                 case SyntaxKind.SemicolonToken:
-                case SyntaxKind.CloseParenToken:
```

Comment:
this is the other half.  we don't mix 'close paren' into the 'bail out set' for error recovery.